### PR TITLE
Don’t auto-select a pullquote text colour for regular style

### LINKED ISSUE #10568: Pull Quote: Selecting Red & Black as main color changes text color, too.
**Describe the bug**
**Gutenberg 4.0 RC** 
The new color options are great for the Pull quote and work well. except when selecting red or black as main color. Both color choices also change text color to a light gray. 

Note: Once you select a color for the text, selecting a red and black doesn't change it any more. Once your clear both colors and start again, you can reproduce it again. 

Tested consistently on Firefox and Google (see below

**To Reproduce**
1. Add a pull quote block
2. Select any main color to see the lines changes. 
3. select the red or the black color and you see both change the lines and the text.

**Expected behavior**

**Screenshots**
Video: http://recordit.co/D22OVpl4LH

**Desktop (please complete the following information):**
 - Device: MacPro- OS: 10.13.6
 - Browser Google / Firefox Developer
 - Version  69.0.3497.100 / 63.0b8

### PR DESCRIPTION
**Pull request by johngodley**
The pullquote block auto-selects an appropriate text colour based on the main colour. This is designed to give a good contrast. However it assumes the quote has a 'solid colour' style where the main colour is the background, with the text overlayed:

edit_post_ _wordpress_latest_ _wordpress

If the pullquote uses the 'regular colour' style the background is white (or whatever the theme uses), and the main colour changes the top and bottom border:

edit_post_ _wordpress_latest_ _wordpress

The colour calculated for the main colour background can often give a poor contrast ratio on this white background.

For example, if you create a pull quote with regular style and pick red as the main colour it sets the text colour to light grey (the calculated text colour for red). In the above solid colour quote this gives good contrast, but on a regular colour quote with white background it gives poor contrast:

edit_post_ _wordpress_latest_ _wordpress

This PR restricts the auto-colour selection to the 'solid colour' style only. I'm not sure if this fits the intended purpose of the block, but it fixes the problem in this instance.

Also fixes the situation where clicking 'clear' on the main colour sets a text colour.

Fixes #10568

## How has this been tested?
1. Create a pull quote block
2. Open block settings
3. Set main colour to red
4. Confirm the text colour does not change
5. Clear the text colour
6. Clear the main colour and verify that the text colour is not set

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/pullquote/edit.js [hunk 1]
```diff
@@ -33,9 +33,12 @@ class PullQuoteEdit extends Component {
...
 
 	pullQuoteMainColorSetter( colorValue ) {
-		const { colorUtils, textColor, setTextColor, setMainColor } = this.props;
+		const { colorUtils, textColor, setTextColor, setMainColor, className } = this.props;
+		const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+		const shouldSetTextColor = isSolidColorStyle && ! textColor.color && colorValue;
+
 		setMainColor( colorValue );
-		if ( ! textColor.color || this.wasTextColorAutomaticallyComputed ) {
+		if ( shouldSetTextColor || this.wasTextColorAutomaticallyComputed ) {
```
Comment (jorgefilipecosta) [inline packages/block-library/src/pullquote/edit.js]: If we go the solid color style we pick the main color, and then we change to the regular style, each time we change the main color the text color continue to change.
I think maybe we can solve this by updating shouldSetTextColor to be `isSolidColorStyle && colorValue && ( ! textColor.color || this.wasTextColorAutomaticallyComputed );`

**Review by jorgefilipecosta — COMMENTED** [review]
Thank you for this change @johngodley this generally works well on my tests 👍  There is only an edge case that I think we may be able to improve.

File: packages/block-library/src/pullquote/edit.js (hunk 1 above)
Comment (johngodley) [reply packages/block-library/src/pullquote/edit.js]: Ah yes, nice! Updated

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

**Pull request merged**